src/optimizer/query_rewrite.c

```diff
@@ -1844,6 +1850,61 @@ qo_reduce_equality_terms (PARSER_CONTEXT * parser, PT_NODE * node, PT_NODE ** wh
       *orgp = parser_append_node (join_term_list, *orgp);
     }
 
+  /* remove always-true term */
+  while ((expr = ((expr_prev) ? expr_prev->next : *orgp)))
+    {
+      PT_OP_TYPE op = expr->info.expr.op;
+      cut_off = false;
+      opd1 = expr->info.expr.arg1;
+      opd2 = expr->info.expr.arg2;
+
+      if (expr->or_next == NULL)
+	{
+	  if (opd1 && opd2 && op == PT_EQ && opd1->node_type == PT_VALUE && opd2->node_type == PT_VALUE)
+	    {
+	      domain = pt_xasl_node_to_domain (parser, expr);
+	      qualifier = expr->info.expr.qualifier;
+	      dbv1 = pt_value_to_db (parser, opd1);
+	      dbv2 = pt_value_to_db (parser, opd2);
+	      if (pt_evaluate_db_value_expr
+		  (parser, expr, expr->info.expr.op, dbv1, dbv2, NULL, &dbval_res, domain, opd1, opd2, NULL, qualifier))
+		{
+		  if (DB_VALUE_TYPE (&dbval_res) == DB_TYPE_INTEGER && db_get_int (&dbval_res) == 1)
+		    {
+		      cut_off = true;
+		    }
+		}
```

Comment:
This is only for PT_EQ, but I think calling the pt_evaluate_db_value_expr function is excessive.
```suggestion
	      dbv1 = pt_value_to_db (parser, opd1);
	      dbv2 = pt_value_to_db (parser, opd2);
	      if (db_value_compare (dbv1, dbv2) == DB_EQ)
		{
		  cut_off = true;
		}
```